src/main/java/edu/wpi/grip/ui/pipeline/input/NumberSpinnerInputSocketView.java

```diff
@@ -29,7 +29,7 @@ public NumberSpinnerInputSocketView(EventBus eventBus, InputSocket<Number> socke
         super(eventBus, socket);
 
 
-        Object[] domain = socket.getSocketHint().getDomain();
+        Object[] domain = socket.getSocketHint().getDomain().orElseThrow(() -> new IllegalStateException("No domain was supplied"));
```

Comment:
Don't throw an exception if the domain isn't specified - spinners don't need a domain, so this will cause some operations to crash